dali/imgcodec/util/convert.h

```diff
@@ -206,20 +206,9 @@ void ApplyOrientation(Orientation orientation, T *&data,
    * direction) and move the data pointer to the end of the axis.
    */
 
-  bool swap_xy = false, flip_x = false, flip_y = false;
-
-  if (orientation.rotate == 90) {
-    swap_xy = true;
-    flip_x = true;
-  } else if (orientation.rotate == 180) {
-    flip_x = true;
-    flip_y = true;
-  } else if (orientation.rotate == 270) {
-    swap_xy = true;
-    flip_y = true;
-  }
-  flip_x ^= orientation.flip_x;
-  flip_y ^= orientation.flip_y;
+  bool swap_xy = orientation.ShouldSwapXYConsideringRotation();
+  bool flip_x = orientation.ShouldFlipXConsideringRotation();
+  bool flip_y = orientation.ShouldFlipYConsideringRotation();
```

Comment:
See the comment in `orientation.h` - I think this is a job of the implementation of the transform to even know if something like "swap_xy" makes sense rather than a part of the definition of the orientation.